auth_password_settings/controllers/main.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# © 2016 Hans Henrik Gabelgaard www.steingabelgaard.dk
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import werkzeug
+import operator
+import logging
+import openerp
+from openerp import http
+from openerp.http import request
+from openerp.tools.translate import _
+from openerp.exceptions import ValidationError, AccessDenied
+from openerp.addons.web.controllers.main import Session
+from openerp.addons.auth_signup.controllers.main import AuthSignupHome
+from openerp.addons.auth_signup.res_users import SignupError
+
+
+_logger = logging.getLogger(__name__)
+
+
+class SessionEx(Session):
+
+    @http.route('/web/session/change_password', type='json', auth="user")
```

Comment:
You are breaking the inheritance chain here. It seems that it might be possible to perform checks in an override before or after calling super. Did you look into that?